core/src/main/java/org/pentaho/di/i18n/GlobalMessageUtil.java

```diff
@@ -0,0 +1,445 @@
+   *
+   * @param packagePath   The package to search in
+   * @param resourceClass the class to use to resolve the bundle
+   * @return The resource bundle
+   * @throws MissingResourceException in case both resource bundles couldn't be found.
+   */
+  public static ResourceBundle getBundle( final String packagePath, final Class<?> resourceClass )
+    throws MissingResourceException {
+    final Set<Locale> activeLocales = getActiveLocales();
+    for ( final Locale locale : activeLocales ) {
+      try {
+        return getBundle( locale, packagePath, resourceClass );
+      } catch ( MissingResourceException e ) {
+        final StringBuilder msg = new StringBuilder();
+        msg.append( "Unable to find properties file for package '" ).append( packagePath ).append( "' and class '" )
+          .append( resourceClass.getName() ).append( "' in the available locales: " ).append( locale );
+        // nothing to do, an exception will be thrown if no bundle is found
+        log.warn( msg.toString() );
+      }
+    }
+    final StringBuilder msg = new StringBuilder();
+    msg.append( "Unable to find properties file for package '" ).append( packagePath ).append( "' and class '" )
+      .append( resourceClass.getName() ).append( "' in the available locales: " ).append(
+      Arrays.asList( activeLocales ) );
+    throw new MissingResourceException( msg.toString(), resourceClass.getName(),
+      packagePath );
+  }
+
+  public static ResourceBundle getBundle( Locale locale, String packagePath, Class<?> resourceClass ) {
+    return getBundle( locale, packagePath, resourceClass, true );
+  }
+
+  /**
+   * Returns a {@link ResourceBundle} corresponding to the given {@link Locale} package and resource class. Falls-back
+   * on the ROOT {@link Locale}, if the {@code fallbackOnRoot} flag is true and the requested Locale is not available.
+   *
+   * @param locale         the {@link Locale} for which the {@link ResourceBundle} is being requested
+   * @param packagePath
+   * @param resourceClass
+   * @param fallbackOnRoot if true, and a {@link ResourceBundle} cannot be found for the requested {@link Locale}, falls
+   *                       back on the ROOT {@link Locale}
+   * @return a {@link ResourceBundle} corresponding to the given {@link Locale} package and resource class
+   */
+  public static ResourceBundle getBundle( final Locale locale, final String packagePath, final Class<?> resourceClass,
+                                          final boolean fallbackOnRoot ) {
+    final GlobalMessageControl control = new GlobalMessageControl( fallbackOnRoot );
+    final String resourceName = control.toResourceName( control.toBundleName( packagePath, locale ), "properties" );
+
+    ResourceBundle bundle;
+    try {
+      bundle = ResourceBundle.getBundle( packagePath, locale, resourceClass.getClassLoader(),
+        new GlobalMessageControl( fallbackOnRoot ) );
+    } catch ( final MissingResourceException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)